branchpro/tests/test_sliders.py

```diff
@@ -0,0 +1,66 @@
+#
+# This file is part of BRANCHPRO
+# (https://github.com/SABS-R3-Epidemiology/branchpro.git) which is released
+# under the BSD 3-clause license. See accompanying LICENSE.md for copyright
+# notice and full license details.
+#
+
+import unittest
+
+import branchpro as bp
+
+
+class Test_SliderComponent(unittest.TestCase):
+    """
+    Test the '_SliderComponent' class.
+    """
+    def test__init__(self):
+        bp._SliderComponent()
+
+    def test_add_slider(self):
+        sliders = bp._SliderComponent()
+        sliders.add_slider('param1', '1', 0, 0, 1, 0.5)
+        self.assertEqual(sliders._sliders[0].children, 'param1')
+        self.assertEqual(sliders._sliders[1].id, '1')
+        self.assertEqual(sliders._sliders[1].min, 0)
+        self.assertEqual(sliders._sliders[1].max, 1)
+        self.assertEqual(sliders._sliders[1].value, 0)
+        self.assertEqual(
+            sliders._sliders[1].marks,
+            {str(ri): str(ri) for ri in [0.0, 0.5, 1.0]}
+            )
+        self.assertEqual(sliders._sliders[1].step, 0.5)
+
+    def test_group_sliders(self):
+        sliders = bp._SliderComponent()
+        sliders.add_slider('param1', '1', 0, 0, 1, 0.5)
+        sliders.add_slider('param2', '2', 0.5, 0, 1, 0.25)
+        sliders.group_sliders()
+
+        self.assertEqual(sliders._sliders[0].children, 'param1')
```

Comment:
Here I would try to test the public interface (i.e. run the asserts on what is returned by the group_sliders() call, or whatever we might rename it to)

Reply:
I am not quite sure what you mean by that. It produces an html object which does not do well with asserts. I checked with David that it is indeed so.

Basically, because in the assert the html objects produced will be non-identical even if they are constructed the same, the test will fail. Also there was no obvious function for asserting equality of two html objects.

Reply:
You are exactly right about trying to check equality of those objects, but what I mean is something like this
```
div = sliders.get_sliders_div()
self.assertEqual(div.children[0].children, 'param1')
self.assertEqual(div.children[1].id, '1')
self.assertEqual(div.children[1].min, 0)
etc
```
If you can get this to work, I believe it will make for a significantly better test, as you are directly checking the expected behavior of the method, namely that it returns this object with the things you expect in it.

Reply:
ok